### PR TITLE
Track better inventory

### PR DESCRIPTION
**Pull request by kknd113**
@curvo @schwartzdev 
First stab at better inventory_advice requests.
Please review when you get the chance.

### CODE REVIEW

**Comment by schwartzdev** [comment]
Looks good. Let's look back 48.hours

**Comment by kknd113** [comment]
This is probably buggy.
Need to squash duplicate keys on `unsynced_variants` hash

File: lib/spree/retailops/rop_stock_helper.rb [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+        
+        qty  
+      end
+      
+      #Returns a hash of variants and its quantities from unsynced orders
+      #key=variant_id
+      #value=quantity
+      #{variant_id => quantity}
+      def get_unsynced_variants
+        arr = Spree::Order.joins(:line_items).where('completed_at > ? AND retailops_import = ?', Time.now-12.hours,'yes').pluck(:variant_id, :quantity)
```
Comment (curvo) [inline lib/spree/retailops/rop_stock_helper.rb]:
```
arr = Spree::LineItem.select('variant_id, SUM(quantity) as quantity').joins(:order).where('completed_at > ? AND retailops_import = ?', Time.now-12.hours,'yes').group(:variant_id).map{|li| [li.variant_id, li.quantity]}
```

**Comment by kknd113** [comment]
Thanks for the review.
1. Fixed AR query.
2. Time window increased to 24 hours.
3. Added edge case checks

Tested on production/staging console.

File: lib/spree/retailops/rop_stock_helper.rb [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+        
+        qty  
+      end
+      
+      #Returns a hash of variants and its quantities from unsynced orders
+      #key=variant_id
+      #value=quantity
+      #{variant_id => quantity}
+      def get_unsynced_variants
+        arr = Spree::LineItem.select('variant_id, SUM(quantity) as quantity').joins(:order).where('completed_at > ? AND retailops_import = ?', Time.now-48.hours,'yes').group(:variant_id).map{|li| [li.variant_id, li.quantity]}
```
Comment (schwartzdev) [inline lib/spree/retailops/rop_stock_helper.rb]: Do we need to memoize this result?

**Comment by kknd113** [comment]
@schwartzdev memoization implemented.

**Comment by kknd113** [comment]
squashed/merged

**Pull request closed without merging**